pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -44,10 +45,6 @@ var (
 	clientLog = log.New(loggerName)
 )
 
-var newDatasourceHttpClient = func(httpClientProvider httpclient.Provider, ds *DatasourceInfo) (*http.Client, error) {
```

Comment:
This was being overridden for tests. Because it relies on the `httpclient.Provider` interface, we can just inject a fake one and rely on the state stored in `baseClientImpl`